pcdsdevices/mv_interface.py

```diff
@@ -19,9 +20,89 @@
 from . import utils as util
 
 logger = logging.getLogger(__name__)
+engineering_mode = True
 
 
-class MvInterface:
+class BaseInterface:
+    """
+    Interface layer to attach to any Device for SLAC features.
+
+    This class defines an API and some defaults for blacklisting tab-completion
+    results for new users to avoid confusion. The API involves setting the
+    following class variables on any subclasses:
+
+    tab_blacklist: list of string regex to omit from tab complete
+    tab_whitelist: list of string regex to include even if blacklisted by a
+                   parent class
+    """
+    tab_blacklist = ["OphydAttrList", "SUB_*", "_*", "actuate",
```

Comment:
```suggestion
    tab_blacklist = ["OphydAttrList", "SUB_.*", "_.*", "actuate",
```

Reply:
I'm going to drop the leading underscore blacklist because ipython automatically hides these unless you insist on doing `device._` then tab